### PR TITLE
1. modify Eiffel Tower effect. 2. translation update. 3. when unit is destroyed the unit table should be not visible.

### PR DESCRIPTION
**Pull request by lishaoxia1985**

### CODE REVIEW

**Comment by lishaoxia1985** [comment]
@JackRainy it's not very beautiful. we can touch the button in the right bottom instead.

**Comment by JackRainy** [comment]
> we can touch the button in the right bottom instead.

@lishaoxia1985 , that's exactly what I call: a bad usability.
Currently, I do just one click to pick the promotion or improvement - it applies and the dialog closes.
And you suggest to do one more extra click: first to click and choose the improvement, and secondly, move to the corner of the screen and hit a tiny button there.
It is just a waste of time.

Even more, "pick now" helps to understand which promotion can be selected right now easily, instead of comparing the colors. Have you seen the report when a person does not understand that policies can be chosen? It can become the same problem here.

**Comment by lishaoxia1985** [comment]
you are right the promotion can be selected should be different color, but picknow is not a good choice.

**Comment by lishaoxia1985** [comment]
you are right the promotion can be selected should be different color, but picknow is not a good choice.

**Comment by lishaoxia1985** [comment]
you are right,  but pick now is not a good choice.
________________________________

**Comment by yairm210** [comment]
https://civilization.fandom.com/wiki/Marble_(Civ5)

The marble effect limitation is only in BNW, not in vanilla or G&K

**Comment by lishaoxia1985** [comment]
so what you mean?  i think we should update a newest civ v branch.
if not, updating the newest at once is a hard job. in fact, i think the BNW is right because at other era the wonder may don't need marble such as the Eiffel Tower.

**Comment by yairm210** [comment]
We're currently aiming for all Vanilla features to work before we branch off the different rulesets.
So there's no need to change anything that conforms to it yet, because in the end they'll all be supported.

**Comment by lishaoxia1985** [comment]
@yairm210 although i don't understand your plan, but i respect your choice.
some days ago i update some policy which has different effect in different branch civ.

**Comment by yairm210** [comment]
From what I see here, you're still changing the marble effect, which is what we don't want =/

**Comment by lishaoxia1985** [comment]
ok,  Give me some times. I will separate it.😂

**Comment by yairm210** [comment]
No rush :)

**Comment by lishaoxia1985** [comment]
@yairm210 👌！！！😄

File: core/src/com/unciv/logic/civilization/CivInfoStats.kt [hunk 1]
```diff
@@ -125,10 +125,10 @@ class CivInfoStats(val civInfo: CivilizationInfo){
             }
         }
 
-        if (civInfo.containsBuildingUnique("Provides 1 happiness per social policy")) {
+        if (civInfo.containsBuildingUnique("Provides 1 happiness per 2 additional social policies adopted")) {
             if(!statMap.containsKey("Policies")) statMap["Policies"]=0f
             statMap["Policies"] = statMap["Policies"]!! +
-                    civInfo.policies.getAdoptedPolicies().count { !it.endsWith("Complete") }.toFloat()
+                    civInfo.policies.getAdoptedPolicies().count { !it.endsWith("Complete") } / 2f
```
Comment (JackRainy) [inline core/src/com/unciv/logic/civilization/CivInfoStats.kt]: As I have already mentioned before, this change for `getAdoptedPolicies().count` = 5 will provide 2.5 happiness, instead of 2.0 as it says "1 happiness per 2 additional social policies".

File: core/src/com/unciv/ui/worldscreen/unit/UnitTable.kt [hunk 2]
```diff
@@ -156,11 +156,7 @@ class UnitTable(val worldScreen: WorldScreen) : Table(){
             selectedUnitHasChanged = true
         }
         else {
-            separator.isVisible=false
-            unitNameLabel.setText("")
-            unitDescriptionTable.clear()
-            unitIconHolder.clear()
-            promotionsTable.clear()
```
Comment (JackRainy) [inline core/src/com/unciv/ui/worldscreen/unit/UnitTable.kt]: Is this change related to the Effel Tower effect?

File: build.gradle [hunk 3]
```diff
@@ -18,7 +18,7 @@ buildscript {
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
         classpath 'de.richsource.gradle.plugins:gwt-gradle-plugin:0.6'
-        classpath 'com.android.tools.build:gradle:3.6.0'
+        classpath 'com.android.tools.build:gradle:3.6.1'
```
Comment (JackRainy) [inline build.gradle]: Why do we need to increment the Gradle version for this change?

File: android/assets/jsons/translationsByLanguage/Traditional_Chinese.properties [hunk 4]
```diff
@@ -535,11 +536,11 @@ Receive triple Gold from Barbarian encampments and pillaging Cities. Embarked un
 
 Spain = 西班牙
 Isabella = 伊莎貝拉一世
-100 Gold for discovering a Natural Wonder (bonus enhanced to 500 Gold if first to discover it). Culture, Happiness and tile yields from Natural Wonders doubled. = 每發現一座自然奇觀時獲得100金錢獎勵(若為第一個發現該自然奇觀的文明，獎勵則追加至500)。自然奇觀所提供的文化、快樂及地區產出雙倍。
+100 Gold for discovering a Natural Wonder (bonus enhanced to 500 Gold if first to discover it). Culture, Happiness and tile yields from Natural Wonders doubled. = 每發現一座自然奇觀時獲得100金錢獎勵(若為第一個發現該自然奇觀的文明，獎勵則追加至500)，自然奇觀所提供的文化、快樂及地區產出雙倍
 
 Mongolia = 蒙古
 Genghis Khan = 成吉思汗
-Combat Strength +30% when fighting City-State units or attacking a City-State itself. All mounted units have +1 Movement. = 進攻城邦及城邦單位時獲得 30% 戰鬥力加成。騎乘單位移動力+1。
+Combat Strength +30% when fighting City-State units or attacking a City-State itself. All mounted units have +1 Movement. = 進攻城邦及城邦單位時獲得30%戰鬥力加成，騎乘單位移動力+1
```
Comment (JackRainy) [inline android/assets/jsons/translationsByLanguage/Traditional_Chinese.properties]: Why do you change this translation? It has nothing to the Effel Tower effect.

Comment (JackRainy) [reply android/assets/jsons/translationsByLanguage/Traditional_Chinese.properties]: The same question - to all other altered translations in this file.

Comment (lishaoxia1985) [reply android/assets/jsons/translationsByLanguage/Traditional_Chinese.properties]: some translation update.

File: build.gradle (hunk 3 above)
Comment (lishaoxia1985) [reply build.gradle]: it uses the latest android studio.

File: core/src/com/unciv/ui/worldscreen/unit/UnitTable.kt (hunk 2 above)
Comment (lishaoxia1985) [reply core/src/com/unciv/ui/worldscreen/unit/UnitTable.kt]: when unit is destroyed the unit table should be not visible.

**Comment by yairm210** [comment]
It looks ok, but please, *please* split your pr's by subject
Having one long pr increases time-to-merge, conflates issues, makes the thread hard to follow and in general is very annoying

File: core/src/com/unciv/logic/civilization/CivInfoStats.kt [hunk 5]
```diff
@@ -125,10 +125,10 @@ class CivInfoStats(val civInfo: CivilizationInfo){
             }
         }
 
-        if (civInfo.containsBuildingUnique("Provides 1 happiness per social policy")) {
+        if (civInfo.containsBuildingUnique("Provides 1 happiness per 2 additional social policies adopted")) {
             if(!statMap.containsKey("Policies")) statMap["Policies"]=0f
             statMap["Policies"] = statMap["Policies"]!! +
-                    civInfo.policies.getAdoptedPolicies().count { !it.endsWith("Complete") }.toFloat()
+                    civInfo.policies.getAdoptedPolicies().count { !it.endsWith("Complete") } / 2
```
Comment (lishaoxia1985) [inline core/src/com/unciv/logic/civilization/CivInfoStats.kt]: now i think it's good.

**Comment by lishaoxia1985** [comment]
OK, I will pay attention to this problem.
